pkg/apis/pravega/v1beta1/pravegacluster_types.go

```diff
@@ -1095,3 +1126,30 @@ func (p *PravegaCluster) NewEvent(name string, reason string, message string, ev
 	}
 	return &event
 }
+
+func (p *PravegaCluster) NewApplicationEvent(name string, reason string, message string, eventType string) *corev1.Event {
```

Comment:
Cant we reuse the NewEvent function?

Reply:
No since that funtion creates event corresponding to the PravegaCluster resource whereas this event is supposed to be generated after the deletion of that resource